app/Libraries/Markdown/Osu/DocumentProcessor.php

```diff
@@ -263,6 +269,15 @@ private function proxyImage()
         }
     }
 
+    private function recordExcerpt(): void
+    {
+        if ($this->excerpt !== null || !$this->node instanceof Paragraph || !$this->event->isEntering()) {
```

Comment:
with first two paragraphs may not even contain actual content I wonder if it's better to just show first 100 characters instead.